C4.puml

```diff
@@ -1161,49 +1161,136 @@ $elementSkin
 !global $propTableCaption = ""
 !global $propColCaption = "="
 
+!global $isFirstProp = 1
+!global $firstPropCol = 1
+!global $lastPropCol = 1
+
+!function $fillMissing($col, $colNext)
+  !if ($col == "" && $colNext != "")
+    !return " "
+  !endif
+  !return $col 
+!endfunction
+
+!function $updatePropColumns($colIdx)
+  !if ($isFirstProp == 1 && $colIdx > $firstPropCol)
+    !$firstPropCol = $colIdx
+  !endif    
+  !if ($isFirstProp == 0 && $colIdx > $lastPropCol)
+    !$lastPropCol = $colIdx
+  !endif    
+  !return "" 
+!endfunction
+
+' add missing header columns, if a following row has more columns
+' (fixed in PlantUML v1.2025.1beta9; only required in older versions)
+!function $fixHeaderColumns()
+  ' the number of displayed columns considers only the first row
+  ' if another row has more columns the first has to be filled with missing columns 
+  !if ($lastPropCol > $firstPropCol)
+    !$delta = $lastPropCol - $firstPropCol
+    !$delta = $delta * 2
+    !$fix = %substr(" | | | |", 0, $delta)
+
+    ' basically the line break \n should be the split
+    ' but \n is not encoded (anymore?) therefore split only via
+    ' \ and remove the last obsolete  \ (changed order with add 
+    ' \ at the beginning is not working). 
+    ' "\n" would split \ and n ==> n would be an unwanted line break 
+    !$lines = %splitstr($propTable, "\")
+    ' !$lines = %splitstr_regex($propTable, "(?=[\x000A])")
```

Comment:
Comment could be removed?

Reply:
in "old" plantuml server \n was encoded as one character, but starting with %newline(), %breakline() \n is encoded as 2  characters \ n with 2 line breaks; therefore I used only \ as line break (and hope that nobody wants to use \\ as escaped \)